MyNotes/Form1.cs

```diff
@@ -46,19 +38,33 @@ private void mainForm_Load(object sender, EventArgs e)
         /// <param name="e"></param>
         private void buttonCreate_Click(object sender, EventArgs e)
         {
+            const int MAX_ATTEMPTS = 10;
+
             textBoxNote.Text = "";
-            if (File.Exists(path + tag)) //если файл Unnamed.txt существует
-            {
-                File.Create(path + random.Next(0, 10).ToString() + tag); //создаём файл с рандомным числом в названии
-            }
-            else //иначе
+
+            string fileName = path + tag;
+            int maxValue = 10;      // --- 
+            int currentAttempt = 0; // --- we want to limit number of attempts trying to find a proper file name
+
+            while (File.Exists(fileName))
             {
+                fileName = $"{path}{random.Next(0, maxValue)}{tag}";
+                maxValue += 10;
 
-                File.Create(path + tag); //создаём файл Unnamed.txt
+                if (currentAttempt++ > MAX_ATTEMPTS)
+                {
+                    MessageBox.Show($"Cannot generate a new file name after {MAX_ATTEMPTS} attempts");
+                    return;
+                }
             }
+
+            FileStream fs = File.Create(fileName); //  создаём файл Unnamed.txt
```

Comment:
You need keep in mind to release "unmanaged" resources, like file descriptors, db connections and etc. to avoid leaks

usually you do it by using the opening inside "using" section, like

```
using(FileStream fs = File.Create(fileName))
{
   fs.Write(...);
}
```

but in this case you do nothing, so you could use empty "using"

```
using(FileStream fs = File.Create(fileName))
{
}
```

or explicitly dispose the resource.

Another option would be to use

` File.WriteAllText(fileName, "");`

i.e. write empty string to the file

Reply:
https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/unmanaged